### PR TITLE
New workloads

### PR DESCRIPTION
**Pull request by pviladegut**
## Changes introduced with this PR

Added iomix and sock stressors 

---
By contributing to this repository, I agree to the [contribution guidelines](https://github.com/arcalot/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by webbnh — COMMENTED** [review]
Looks good, although handling each stressor individually is sort of tedious (not your fault!).  I've submitted PR https://github.com/pviladegut/arcaflow-plugin-stressng/pull/1 to your branch to remedy that, if you are interested in addressing it.

In addition to the typographical comments below, you should run Black on the code to format it (I've done that in my PR).

**Comment by webbnh** [comment]
FWIW, I've addressed all my comments here in my PR to your branch, in case that's helpful.  🙃

**Review by webbnh — CHANGES_REQUESTED** [review]
I note that there are tests for the existing stressors ([folder](https://github.com/arcalot/arcaflow-plugin-stressng/tree/7d4764ab6435c402a25cade17426c17acc10ea0e/tests)).  @pviladegut, can you please add tests for the stressors that you're adding?

**Comment by dustinblack** [comment]
> Looks good, although handling each stressor individually is sort of tedious (not your fault!). I've submitted PR [pviladegut#1](https://github.com/pviladegut/arcaflow-plugin-stressng/pull/1) to your branch to remedy that, if you are interested in addressing it.
> 
> In addition to the typographical comments below, you should run Black on the code to format it (I've done that in my PR).

This is a valuable improvement (thanks @webbnh!), but it feels out-of-context with Patricia's PR and I question whether it should be a separate PR to this repo directly.

**Comment by webbnh** [comment]
> it feels out-of-context with Patricia's PR and I question whether it should be a separate PR to this repo directly.

If we take as given that it is out-of-context, should it go in ahead of Patricia's PR or after it?  😈

Since using the table is sort of an "infrastructure" change which should have no functional impact (other than, perhaps, performance), I would suggest that it is foundational and should go in first.  I'm willing to execute that, if you concur.

Otherwise, her change adds yet another workload, which is what adopting the table-driving approach is supposed to make easy and encapsulated...so, there is not a lot to be gained by deferring adding the table to after expanding its domain.

Also, I think we're waiting on testing; so presumably there is time to do the table-conversion first.

**Comment by dustinblack** [comment]
> > it feels out-of-context with Patricia's PR and I question whether it should be a separate PR to this repo directly.
> 
> If we take as given that it is out-of-context, should it go in ahead of Patricia's PR or after it? 😈
> 
> Since using the table is sort of an "infrastructure" change which should have no functional impact (other than, perhaps, performance), I would suggest that it is foundational and should go in first. I'm willing to execute that, if you concur.
> 
> Otherwise, her change adds yet another workload, which is what adopting the table-driving approach is supposed to make easy and encapsulated...so, there is not a lot to be gained by deferring adding the table to after expanding its domain.
> 
> Also, I think we're waiting on testing; so presumably there is time to do the table-conversion first.

I agree; I would do the table conversion first, and then rebase and re-work this PR from @pviladegut.

**Comment by webbnh** [comment]
> I would do the table conversion first

Done:  #42.  (`git cherry-pick`...and some merging...ftw!)

> then rebase and re-work this PR from @pviladegut.

It'll be a test, but merging https://github.com/pviladegut/arcaflow-plugin-stressng/pull/1 into Patricia's branch, then squashing, and then rebasing might allow Git to do it all automagically.

**Comment by pviladegut** [comment]
Hi,
sorry about the late reply. I'm not clear about this process, should I
commit the changes I made to my branch or merging pviladegut#1 into my
current branch would be enough?

Thanks
-Patricia

**Comment by webbnh** [comment]
> sorry about the late reply.

No worries -- you are providing the code, largely for your own benefit, so I think we're happy to operate on your schedule.  🙂

> I'm not clear about this process

Nor am I! 😉  Dustin has suggested (and I don't disagree) that my refactoring to make the stressors management be table-driven should go in before your changes.  So, I've posted #42, and, once that's merged, you can bring your branch up to date with that state.  There are two ways to do that, `git rebase` and `git merge`.  Either way, if you try to update your branch as it is without first merging [pviladegut#1](https://github.com/pviladegut/arcaflow-plugin-stressng/pull/1), you are likely to have unpleasant merge-conflicts to resolve (like, possibly, in _every one of your commits_, if you rebase).  (Note that you can try it, and if you don't like it, you can abort; you can also check out a new branch from your existing branch, which will give you another way to get your branch back if you run into trouble.)

So, my recommendation is to merge [pviladegut#1](https://github.com/pviladegut/arcaflow-plugin-stressng/pull/1) into your branch, which will mean that your branch already contains the changes which #42 will put on to `new-workloads` (although, it looks like Jared is requesting some tweaks, which will slightly complicate things for you).  Once my changes are in both `new-workloads` and on your branch, then, if you merge `new-workloads` into your branch, it should be fairly easy, because the source code structure will match and Git should handle most of the details.  However, if you are inclined (as I normally am) toward using `git rebase` instead of `git merge` to join your branch with `new-workloads`, Git will likely force you to resolve merge-conflicts for each commit, which as I said above, will be ugly; however, you can avoid that by squashing your branch after you merge [pviladegut#1](https://github.com/pviladegut/arcaflow-plugin-stressng/pull/1) into it:  this will mean you have only one commit to rebase, and that commit will already have the new structure, and Git will hopefully take care of most of the details with a minimum of fuss (modulo the changes I make on `new-workloads` for Jared).

So, to sum up, we are going to merge #42 while you work on adding testing; whenever you are ready, you should merge [pviladegut#1](https://github.com/pviladegut/arcaflow-plugin-stressng/pull/1) into your branch; once that's done and #42 is merged, you should either merge `new-workloads` into your branch or squash your branch and rebase it on `new-workloads` (either way, you'll probably have some merge conflicts to resolve, but they should be small and easy).  And, then you can push your updated branch, and we'll review it again.  (And, the diffs should be relatively small, thanks #42.)

Let me know if that's unclear or if you have any questions or if you need any help.

**Review by dustinblack — APPROVED** [review]

File: arcaflow_plugin_stressng/stressng_schema.py [hunk 1]
```diff
@@ -30,6 +30,12 @@ class Stressors(str, enum.Enum):
...
     IOMIX = "iomix"
+    SOCK = "sock"
+
+
+# Mapping of Stressors to their corresponding output schemas (each schema
+# definition is added as it is defined, below).
+stressor_schemas = {}
 
 
 # Mapping of Stressors to their corresponding output schemas (each schema
```
Comment (webbnh) [inline arcaflow_plugin_stressng/stressng_schema.py:41]: Lines 36-40 are duplicated starting at line 41.

**Review by webbnh — COMMENTED** [review]
Merging is tricky.  🥴
